[FEATURE] Registers a student to the platform

Njaya2019 opened the pull request:
## Description
It adds a student to the platform.

### client request
```
POST users/students/register
```
### request body
```
{
    "firstname": "Weclif",
    "middlename": "Mzae",
    "lastname": "Kisao",
    "Address": "Mombasa, Kenya",
    "email": "willytuvaK0313@gmail.com",
    "age": 13,
    "educationlevel": "Olevel"
}
```
### successful response example
```
{
    "id": 1,
    "firstname": "Willy",
    "middlename": "Mzae",
    "lastname": "Kisao",
    "Address": "Mombasa, Kenya",
    "email": "willymzae@gmail.com",
    "age": 13,
    "educationlevel": "Olevel"
}
```
### successful response type and code
```
 response type: application/json
 status code: 201
```
### Dependencies
```
- Django==3.0.6
- djangorestframework==3.11.0
- pylint==2.5.2
- coverage==4.4
- pytest==5.4.2
- pytest-cov==2.8.1
- pytest-django==3.9.0
- pytest-env==0.6.2
```
**Note** tests won't run if *coverage* is a latest version
All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #(issue number #4 )

## How Has This Been Tested?
**Clone**
```
git clone --single-branch --branch ft-registers-a-student  git@github.com:BuildForSDG/team-195.git
```
**Installation**
change directory to the tutor folder
```
team-195> cd tutor
```
Install all dependencies first
```
pip install -r requirements/local.txt 
pip install -r requirements/base.txt 
```
Set environment variables for both test and base database.
```
DATABASE_URL=psql://USER:PASSWORD@HOST:HOST/your-database
TEST_DATABASE=psql://USER:PASSWORD@HOST:HOST/test-database
```
Run migrations
```
- python manage.py makemigrations
- python manage.py migrate
```
run
```
python manage.py runserver
```

## Screenshots (if applicable, else remove this line / section)
**Bad request image**
![image](https://user-images.githubusercontent.com/39832798/82024413-6ca26980-9698-11ea-9717-19ba41b3d40f.png)
**Successful request image**
![image](https://user-images.githubusercontent.com/39832798/82024558-a4111600-9698-11ea-93fd-1e628070b233.png)

## Checklist:
<!--- Put an `x` in all the boxes that apply ! -->
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

Njaya2019 committed:
[chore] formated code to conform to PEP 8 code style standard

Njaya2019 committed:
[feature] Added a feature to register a student to the platform

BernardMW commented:
I think this is perfect. However, I would like us to make use of the sign up model that we get by default. What you have very good. But it can still be improved.

BernardMW commented:
I particularly applaud your work on serializers. Great job.

BernardWambua reviewed (dismissed):
This is okay. However I think if you can inherit from the user registration model it will even be better.

peterkuria committed:
updated the list of allowed hosts to connect

This is to allow us to be able to use the development server locally and also when we deploy to heroku.

peterkuria committed:
update timezone, database config and env options.

Added multiple database options and the use of environment variables for our server.

peterkuria reviewed (dismissed):
Great progress. The API endpoint is working well. I incorporated my own feedback in your pull request with a few additional commits. Test passed after reviewing the few bugs such as timezone.

Njaya2019 committed:
[chore] Added database enviroment variables to test and base databases.

Njaya2019 committed:
[chore] Pulled changes from remote repository and modified the code to conform PEP standards

Njaya2019 committed:
[chore] Modified local and base settings to solve import errors

BernardWambua commented:
I think Education level field can be a lookup field.

BernardWambua commented:
@Njaya2019 Kindly change the education level to lookup. It's the only issue I see with this branch the we can merge it.

BernardWambua reviewed (approved):
Approved.

peterkuria commented on tutor/config/settings/base.py:
```diff
@@ -2,21 +2,24 @@
+READ_DOT_ENV_FILE = env("DJANGO_READ_DOT_ENV_FILE", default=False)
+
+# OS environment variables take precedence over variables from .env
+env.read_env(str(ROOT_DIR / ".env"))
```
changed environmental variable clause..

peterkuria reviewed (approved):
Approved changes to db config
